reconcile/test/test_acs_rbac.py

```diff
@@ -0,0 +1,714 @@
+                    "value": "foofoo@redhat.com",
+                },
+            }
+        ),
+        acs_api.Group(
+            api_data={
+                "roleName": "cluster-analyst",
+                "props": {
+                    "id": "4",
+                    "authProviderId": AUTH_PROVIDER_ID,
+                    "key": "email",
+                    "value": "barbar@redhat.com",
+                },
+            }
+        ),
+        acs_api.Group(
+            api_data={
+                "roleName": "service-vuln-admin",
+                "props": {
+                    "id": "5",
+                    "authProviderId": AUTH_PROVIDER_ID,
+                    "key": "email",
+                    "value": "foobar@redhat.com",
+                },
+            }
+        ),
+    ]
+
+
+@pytest.fixture
+def api_response_access_scopes() -> list[acs_api.AccessScope]:
+    return [
+        acs_api.AccessScope(
+            api_data={
+                "id": "1",
+                "name": "Unrestricted",
+                "description": "Access to all clusters and namespaces",
+                "rules": None,
+            }
+        ),
+        acs_api.AccessScope(
+            api_data={
+                "id": "2",
+                "name": "cluster-analyst",
+                "description": "analyst access to clusters in acs instance",
+                "rules": {
+                    "includedClusters": ["clusterA", "clusterB"],
+                    "includedNamespaces": [],
+                },
+            }
+        ),
+        acs_api.AccessScope(
+            api_data={
+                "id": "3",
+                "name": "service-vuln-admin",
+                "description": "vuln-admin access to service namespaces in acs instance",
+                "rules": {
+                    "includedClusters": [],
+                    "includedNamespaces": [
+                        {
+                            "clusterName": "stage-cluster",
+                            "namespaceName": "serviceA-stage",
+                        },
+                        {
+                            "clusterName": "prod-cluster",
+                            "namespaceName": "serviceA-prod",
+                        },
+                    ],
+                },
+            }
+        ),
+    ]
+
+
+@pytest.fixture
+def api_response_permission_sets() -> list[acs_api.PermissionSet]:
+    return [
+        acs_api.PermissionSet(
+            api_data={
+                "id": "1",
+                "name": "Admin",
+            }
+        ),
+        acs_api.PermissionSet(
+            api_data={
+                "id": "2",
+                "name": "Analyst",
+            }
+        ),
+        acs_api.PermissionSet(
+            api_data={
+                "id": "3",
+                "name": "Vulnerability Management Admin",
+            }
+        ),
+    ]
+
+
+def test_get_desired_state(mocker, query_data_desired_state, modeled_acs_roles):
```

Comment:
Since this is a new test file, it would be good to have types in all methods.

Reply:
[updated](https://github.com/app-sre/qontract-reconcile/pull/3899/commits/30c10c560d2dab2db4a5ee098b2c3715ab5c8448)